In [2]:
import pandas as pd
spain_df = pd.read_csv("data/spain.csv")


First we import the data and explore it.

In [2]:
spain_df

Date  Season                home             visitor   HT   FT  \
0      1929-02-10    1928     Arenas de Getxo     Atletico Madrid  0-2  2-3   
1      1929-02-10    1928  Espanyol Barcelona          Real Union  1-0  3-2   
2      1929-02-10    1928         Real Madrid           CE Europa  0-0  5-0   
3      1929-02-10    1928       Real Sociedad     Athletic Bilbao  1-1  1-1   
4      1929-02-12    1928    Racing Santander        FC Barcelona  0-0  0-2   
...           ...     ...                 ...                 ...  ...  ...   
26190  2022-05-22    2021          Granada CF  Espanyol Barcelona  NaN  0-0   
26191  2022-05-22    2021          CA Osasuna        RCD Mallorca  NaN  0-2   
26192  2022-05-22    2021        FC Barcelona       Villarreal CF  NaN  0-2   
26193  2022-05-22    2021          Sevilla FC     Athletic Bilbao  NaN  1-0   
26194  2022-05-22    2021       Real Sociedad     Atletico Madrid  NaN  1-2   

       hgoal  vgoal  tier   round group notes  
0          2      3     1  league   NaN   NaN  
1          3      2     1  league   NaN   NaN  
2          5      0     1  league   NaN   NaN  
3          1      1     1  league   NaN   NaN  
4          0      2     1  league   NaN   NaN  
...      ...    ...   ...     ...   ...   ...  
26190      0      0     1  league   NaN   NaN  
26191      0      2     1  league   NaN   NaN  
26192      0      2     1  league   NaN   NaN  
26193      1      0     1  league   NaN   NaN  
26194      1      2     1  league   NaN   NaN  

[26195 rows x 12 columns]

In [9]:
team_list = spain_df['home'].unique()
team_list


array(['Arenas de Getxo', 'Espanyol Barcelona', 'Real Madrid',
       'Real Sociedad', 'Racing Santander', 'FC Barcelona',
       'Athletic Bilbao', 'Atletico Madrid', 'Real Union', 'CE Europa',
       'CD Alaves', 'Valencia CF', 'Real Betis', 'Real Oviedo',
       'Sevilla FC', 'Hercules CF', 'CA Osasuna', 'Real Zaragoza',
       'Celta Vigo', 'Real Murcia', 'Granada CF', 'Deportivo La Coruna',
       'CD Castellon', 'CE Sabadell', 'Sporting Gijon', 'CD Alcoyano',
       'Gimnastic', 'Real Valladolid', 'CD Malaga', 'UE Lleida',
       'Atletico Tetuan', 'UD Las Palmas', 'Real Jaen', 'CD Leonesa',
       'CD Condal', 'Elche CF', 'RCD Mallorca', 'CD Tenerife',
       'Cordoba CF', 'Pontevedra CF', 'Levante UD', 'Burgos CF',
       'UD Salamanca', 'Cadiz CF', 'Rayo Vallecano', 'Recreativo Huelva',
       'AD Almeria', 'CD Logrones', 'Real Burgos', 'Albacete',
       'SD Compostela', 'CP Merida', 'CF Extremadura', 'Villarreal CF',
       'CD Numancia', 'Malaga CF', 'Getafe CF', 'UD Almeri

In [6]:
spain_df['home'].value_counts()

Real Madrid        1476
FC Barcelona       1476
Athletic Bilbao    1476
Valencia CF        1427
Atletico Madrid    1402
                   ... 
CE Europa            27
Xerez CD             19
CD Condal            15
CD Leonesa           15
Atletico Tetuan      15
Name: home, Length: 63, dtype: int64

In our results above Real Madrid, Barcelona, and Athletic Club Bilbao have played the most games since 1929. It is likely these teams have never been relegated or existed longer than all others.

In [9]:
spain_df['hgoal'].describe()

count    26195.000000
mean         1.802863
std          1.488419
min          0.000000
25%          1.000000
50%          2.000000
75%          3.000000
max         12.000000
Name: hgoal, dtype: float64

On average the home team scored 1.8 goals per game. 
The most any team has scored at home in the leagues is 12 goals

In [10]:
spain_df['vgoal'].describe()

count    26195.000000
mean         1.037793
std          1.084513
min          0.000000
25%          0.000000
50%          1.000000
75%          2.000000
max          8.000000
Name: vgoal, dtype: float64

Visitors score an average of 1.04 goals per game. There is a clear home team advantage.
The most an away team has scored is 8 goals.

In [14]:
spain_df['notes'].unique()

array([nan, '1-0 win awarded to AD Almeria'], dtype=object)

There seems to be one game where the score may have changed after the fact.

In [3]:
spain_df['tier'].describe()

count    26195.0
mean         1.0
std          0.0
min          1.0
25%          1.0
50%          1.0
75%          1.0
max          1.0
Name: tier, dtype: float64

This column indicates that all games recorded are 1st tier games. This is the top professional division in Spain.

In [5]:
spain_df['group'].unique()
spain_df['group'].value_counts()

B    30
C    30
A    30
Name: group, dtype: int64

There seems to be thirty games or so with a group designation. More exploration is needed to undestand these.

In [8]:
spain_df['FT'].describe()

count     26195
unique       79
top         1-0
freq       3064
Name: FT, dtype: object

The most frequent score was 1-0. This happened in 3064 games.

In [16]:
visitor_team_list = spain_df['visitor'].unique()
visitor_team_list.sort()
visitor_team_list

team_list.sort()
team_list

if team_list.all() == visitor_team_list.all():
    print('These lists are the same')
else: 
    print('These lists do not match')



These lists are the same


In [22]:
spain_df['Season'].unique()
spain_df['Season'].value_counts()


1996    462
1995    462
1986    396
2021    380
1994    380
       ... 
1930     90
1929     90
1932     90
1933     90
1928     90
Name: Season, Length: 91, dtype: int64

The 1996 and 1995 seasons had 462 games! Early seasons seem to have only 90 games a season.

380 games seems to be the recent average.